climpred/classes.py

```diff
@@ -152,6 +153,113 @@ def __repr__(self):
         else:
             return _display_metadata(self)
 
+    def _math(self, other, operator):
+        """Helper function for __add__, __sub__, __mul__, __truediv__.
+
+        Allows math operations with type:
+            - int
+            - float
+            - np.ndarray
+            - xr.DataArray without new dimensions
+            - xr.Dataset without new dimensions or variables
+
+        """
+        assert isinstance(operator, str)
+
+        def add(a, b):
+            return a + b
+
+        def sub(a, b):
+            return a - b
+
+        def mul(a, b):
+            return a * b
+
+        def div(a, b):
+            return a / b
+
+        ALLOWED_TYPES_FOR_MATH_OPERATORS = [
+            int,
+            float,
+            np.ndarray,
+            xr.DataArray,
+            xr.Dataset,
+            type(self),
+        ]
+        # catch other dimensions in other
+        error_str = f'Cannot use {type(self)} {operator} {type(other)}'
+        if isinstance(other, tuple([xr.Dataset, xr.DataArray])):
+            all_dims = set(
+                [d for d in list(self._datasets['initialized'].dims)] + CLIMPRED_DIMS
+            )
+            for dim in other.dims:
+                if dim not in all_dims:
+                    raise DimensionError(f'{error_str} containing new dimensions.')
+        # catch xr.Dataset with different data_vars
+        if isinstance(other, xr.Dataset):
+            if list(other.data_vars) != list(self._datasets['initialized'].data_vars):
+                raise VariableError(
+                    f'{error_str} with new `data_vars`. Please use {type(self)} '
+                    f'{operator} {type(other)} only with same `data_vars`. Found '
+                    f'initialized.data_vars = '
+                    f' {list(self._datasets["initialized"].data_vars)} vs. '
+                    f'other.data_vars = { list(other.data_vars)}.'
+                )
+        # catch undefined types for other
+        if not isinstance(other, tuple(ALLOWED_TYPES_FOR_MATH_OPERATORS)):
+            raise TypeError(
+                f'{error_str} because type {type(other)} not supported. '
+                f'Please choose from {ALLOWED_TYPES_FOR_MATH_OPERATORS}.'
+            )
+
+        operator = eval(operator)
+
+        # Create temporary copy to modify to avoid inplace operation.
+        datasets = self._datasets.copy()
+        for dataset in datasets:
+            if isinstance(other, PredictionEnsemble):
+                other_dataset = other._datasets[dataset]
+            else:
+                other_dataset = other
+            if self._datasets[dataset]:
```

Comment:
Add comment `# Some pre-allocated entries might be empty, such as 'uninitialized'.`